### PR TITLE
Bump Go version to 1.22.4

### PR DESCRIPTION
**Pull request by archups**
**What this PR does / why we need it**:
This PR is to update Go version to 1.22.4.
Related to https://github.com/kubermatic/infra/pull/2707

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Update Go version to 1.22.4
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Review by embik — CHANGES_REQUESTED** [review]

File: modules/web/containers/chrome-headless/Dockerfile [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
-    sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' && \
-    apt-get update -qq && \
-    apt-get install -y \
-      google-chrome-stable \
-      libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+  sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' && \
+  apt-get update -qq && \
+  apt-get install -y \
+  google-chrome-stable \
+  libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply modules/web/containers/chrome-headless/Dockerfile:25]: Please revert this change

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
https://github.com/kubermatic/dashboard/blob/main/modules/api/hack/gen-api-client.sh#L27

golang:1.22.2 -> quay.io/kubermatic/build:go-1.22-node-20-9


@archups please make sure that you properly search for usage of golang, `golang`, `1.22.` in the codebase and update all those instances. Seems like we have the same issues in all the PRs.

File: modules/web/containers/chrome-headless/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (archups) [reply modules/web/containers/chrome-headless/Dockerfile]: done!

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**